payments/payments.go

```diff
@@ -0,0 +1,41 @@
+package payments
+
+import (
+	"context"
+	"net/http"
+)
+
+const (
+	StripeProvider = "stripe"
+	PayPalProvider = "paypal"
+)
+
+// Provider represents a payment provider that can optionally charge, refund,
+// preauthorize payments.
+type Provider interface {
```

Comment:
I think what you could do to address the support is just have an error that you could type check. 

```
if err := tx.Charge(...); err != nil {
  if _, ok := err.(NotSupportedError); ok {
    ...
  }
}
```

I think that the concern about controlling access to that data in the code is not that necessary. I think it murks up the code more than it enforces things.